src/components/controller/loader.ts

```diff
@@ -0,0 +1,78 @@
+import { CallbackFunction } from '../types/CallbackFunction';
+
+interface LoaderOptions {
+    apiKey: string;
+}
+
+interface RequestOptions {
+    sources?: string;
+}
+
+interface RequestParams {
+    endpoint: string;
+    options?: RequestOptions;
+}
+
+interface ILoader {
+    baseLink: string;
+    options: LoaderOptions;
+    getResp: <T extends object>(params: RequestParams, callback?: CallbackFunction<T>) => void;
+    errorHandler: (res: Response) => Response;
+    makeUrl: (options: RequestOptions, endpoint: string) => void;
+    load: (method: string, endpoint: string, callback: (data: unknown) => void, options: RequestOptions) => void;
+}
+
+class Loader implements ILoader {
+    baseLink: string;
+    options: LoaderOptions;
+
+    constructor(baseLink: string, options: LoaderOptions) {
+        this.baseLink = baseLink;
+        this.options = options;
+    }
+
+    getResp<T extends object>(
+        { endpoint, options = {} }: RequestParams,
+        callback: CallbackFunction<T> = () => {
+            console.error('No callback for GET response');
+        }
+    ): void {
+        this.load('GET', endpoint, callback, options);
```

Comment:
Recommendation: "GET" and other HTTP Methods could be extracted to Enum and even used as a type in load function

```ts
Enum HTTPMethods {
  Get = "GET",
  Post = "POST",
  // others
}

// Usage as a type:
load<T extends object>(
        method: HTTPMethods,
        endpoint: string,
        callback: CallbackFunction<T>,
        options: RequestOptions = {}
    ) { /* ... */ }
```